Description: Fix catman iconv ordering
 2.5.7 introduced a regression when running catman in some locales, most
 notably in the C locale: while converting the output to UTF-8, iconv was
 run after the compressor rather than before it.  This patch fixes that.
Origin: upstream, http://bazaar.launchpad.net/~cjwatson/man-db/trunk/revision/1220
Bug-Debian: http://bugs.debian.org/593350
Bug-Ubuntu: https://bugs.launchpad.net/bugs/615045
Forwarded: yes
Last-Update: 2010-08-17

Index: b/src/man.c
===================================================================
--- a/src/man.c
+++ b/src/man.c
@@ -2248,12 +2248,13 @@
 	char *tmpcat = tmp_cat_filename (cat_file);
 	int status;
 
+	add_output_iconv (format_cmd, encoding, "UTF-8");
+
 #ifdef COMP_CAT
 	pipeline_command_argstr (format_cmd,
 				 get_def ("compressor", COMPRESSOR));
 #endif /* COMP_CAT */
 
-	add_output_iconv (format_cmd, encoding, "UTF-8");
 	maybe_discard_stderr (format_cmd);
 	format_cmd->want_out = tmp_cat_fd;
 
